crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -517,13 +517,16 @@ impl AuthorityStorePruner {
             "Starting object pruning service with num_epochs_to_retain={}",
             config.num_epochs_to_retain
         );
-        let tick_duration = Duration::from_millis(config.pruning_run_delay_seconds.unwrap_or(
-            if config.num_epochs_to_retain > 0 {
-                min(epoch_duration_ms / 2, 60 * 60 * 1000)
-            } else {
-                min(epoch_duration_ms / 2, 60 * 1000)
-            },
-        ));
+        let tick_duration = Duration::from_millis(match config.pruning_run_delay_seconds {
+            None => {
+                if config.num_epochs_to_retain > 0 {
+                    min(epoch_duration_ms / 2, 60 * 60 * 1000)
+                } else {
+                    min(epoch_duration_ms / 2, 60 * 1000)
+                }
+            }
+            Some(duration_seconds) => duration_seconds * 1000,
```

Comment:
good catch, thankfully this setting was almost never used

Reply:
nice catch!
With watermark safeguards in place and embracing an aggressive pruner, we no longer need this logic. 
Feel free to remove it and simply set the tick interval to some default value (like a minute)